schemas/common.json

```diff
@@ -0,0 +1,109 @@
+{
+  "$schema": "http://json-schema.org/schema#",
+  "$id": "https://github.com/w3c/reffy/blob/main/schemas/common.json",
+
+  "$defs": {
+    "url": {
+      "type": "string",
+      "format": "url"
+    },
+
+    "nullableurl": {
+      "oneOf": [
+        { "$ref": "#/$defs/url" },
+        { "type": "null" }
+      ],
+      "$comment": "Extracts sometimes use null values for URLs, they should probably rather drop the property"
+    },
+
+    "title": {
+      "type": "string",
+      "minLength": 1
+    },
+
+    "shortname": {
+      "type": "string",
+      "pattern": "^[\\w\\-]+((?<=-\\d+)\\.\\d+)?$",
+      "$comment": "Same definition as in browser-specs"
+    },
+
+    "specInExtract": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "title": { "$ref": "#/$defs/title" },
+        "url": { "$ref": "#/$defs/url" }
+      },
+      "required": ["title", "url"]
+    },
+
+    "cssPropertyName": {
+      "type": "string",
+      "minLength": 1
+    },
+
+    "cssValue": {
+      "type": "string",
+      "minLength": 1
+    },
+
+    "interface": {
+      "type": "string",
+      "pattern": "^[A-Z]([A-Za-z0-9_])*|console$",
```

Comment:
this allows any name ending with "console", doesn't it? probably not a big deal though

Reply:
Ah, good point, some parentheses are missing or `^` and `$` need to be repeated. I'll fix that because it also currently allows interfaces that end with clunky characters!